src/endpoint/s3/ops/s3_get_bucket_policy_status.js

```diff
@@ -16,27 +17,17 @@ async function get_bucket_policy_status(req) {
 // TODO: implemented according to current implementation of authorize_request_policy. should update when authorize_request_policy changed
 // full public policy defintion: https://docs.aws.amazon.com/AmazonS3/latest/dev/access-control-block-public-access.html#access-control-block-public-access-policy-status
 function _is_policy_public(policy) {
-    for (const statement of policy.statement) {
-        let principal_wildcard = false;
-        let resource_wildcard = false;
-        if (statement.effect === 'deny') {
-            return false;
-        }
-        for (const principal of statement.principal) {
-            if (principal.unwrap() === '*') {
-                principal_wildcard = true;
-            }
-        }
-        for (const resource of statement.resource) {
-            if ((/[?*]/).test(resource)) {
-                resource_wildcard = true;
+    for (const statement of policy.Statement) {
+        if (statement.Effect === 'Allow' && statement.Principal) {
+            const statement_principal = statement.Principal.AWS ? statement.Principal.AWS : statement.Principal;
+            for (const principal of _.flatten([statement_principal])) {
```

Comment:
is this something possible? that we will have more than one principle, and one of them be '*' (seems redundant no?)

Reply:
it is redundant but possible both in our code and AWS. In AWS if at least one of the principles is * it counts as public policy. so we might as well do the same